alacritty/src/renderer/mod.rs

```diff
@@ -410,17 +410,17 @@ struct InstanceData {
 #[derive(Debug)]
 pub struct QuadRenderer {
     program: TextShaderProgram,
-    rect_program: RectShaderProgram,
     vao: GLuint,
     ebo: GLuint,
     vbo_instance: GLuint,
-    rect_vao: GLuint,
-    rect_vbo: GLuint,
     atlas: Vec<Atlas>,
     current_atlas: usize,
     active_tex: GLuint,
     batch: Batch,
     rx: mpsc::Receiver<Msg>,
+
+    rect_renderer: SolidRectRenderer,
+    rect_program: RectShaderProgram,
```

Comment:
I think we can move `rect_program` into `rect_renderer`? I don't see any reason to not have it there. I also question whether we should have something like and split-out quad renderer into a separate file, since `renderer/mod.rs` is getting big tbqh? Can do it separately though.

```rust
struct Renderer {
    quad_renderer: QuadRenderer,
    rect_renderer: RectRenderer,

    rx: mspc::Receiver<Msg>,
}
```

What do you think @chrisduerr ?

Reply:
A avoided moving `rect_program` into SolidRectRender because program reloading is coupled with toplevel QuadRenderer. I will try to make it work.

Reply:
>I avoided moving rect_program into SolidRectRender because program reloading is coupled with toplevel QuadRenderer. I will try to make it work.

I've explicitly checked that it'll work fine, so you just access a field on a rect render, which is basically the same as accessing anything on quad renderer, since rect one is already there.

Reply:
Done

Reply:
Btw, since we've mentioned splitting QuadRenderer into quad_renderer.rs are you fine with doing so or will do that ourselves?

Also, the file should be `rect_renderer.rs`.

Reply:
Could also just do `renderer/quad.rs` and `renderer/rect.rs`. Of course `rect.rs` already exists, but that would probably be okay since these two things are related.

Reply:
It's `rects.rs`, so we can have both!

If you feel like it should be a part of this PR, I can try to make it.